### PR TITLE
Issue #31: ETW APIs are unimplemented

### PR DESCRIPTION
**Pull request by CBaud**

### CODE REVIEW

File: WindowsDevicePortalWrapper/UnitTestProject/Core/EtwTests.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            ValidateEtwEvents(etwEvents);
+        }
+
+        /// <summary>
+        /// Validate the <see cref="EtwEvents"/>  returned from the tests.
+        /// </summary>
+        /// <param name="etwEvents">The <see cref="EtwEvents"/> to validate.</param>
+        private static void ValidateEtwEvents(EtwEvents etwEvents)
+        {
+            Assert.IsNotNull(etwEvents);
```
Comment (Scottj1s) [inline WindowsDevicePortalWrapper/UnitTestProject/Core/EtwTests.cs]: Don't need to check event list elements?

**Comment by Scottj1s** [comment]
Great work!  Nice to see this support.

File: WindowsDevicePortalWrapper/UnitTestProject/Core/EtwTests.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        /// <summary>
+        /// Basic test of GET method for getting a list of registered ETW providers.
+        /// </summary>
+        [TestMethod]
+        public void GetEtwProvidersTest()
+        {
+            TestHelpers.MockHttpResponder.AddMockResponse(DevicePortal.EtwProvidersApi, HttpMethods.Get);
+
+            Task<EtwProviders> getEtwProvidersTask = TestHelpers.Portal.GetEtwProvidersAsync();
```
Comment (mschofie) [inline WindowsDevicePortalWrapper/UnitTestProject/Core/EtwTests.cs]: I believe that the VS test infrastructure supports making methods 'public async Task' so you can 'await' on the task to make things a little cleaner and more representative; i.e. 
EtwProviders etwProviders = await TestHelpers.Portal.GetEtwProvidersAsync();

File: WindowsDevicePortalWrapper/UnitTestProject/WDPMockImplementations/WebSocket.cs [hunk 3]
```diff
@@ -37,6 +37,11 @@ internal partial class WebSocket<T>
         private Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> serverCertificateValidationHandler;
 
         /// <summary>
+        /// The connection to the websocket.
+        /// </summary>
+        private Task<HttpResponseMessage> webSocketTask;
```
Comment (mschofie) [inline WindowsDevicePortalWrapper/UnitTestProject/WDPMockImplementations/WebSocket.cs]: You're adding an IDisposable member - shouldn't you add an IDisposable implementation to this class? That being said, there was already an IDisposable member, so perhaps it's unnecessary?

**Comment by davidkline-ms** [comment]
This is fantastic! Thanks for implementing this, Chris!

**Pull request merged**